v2/api/apimanagement/v1api20220801/structure.txt

```diff
@@ -206,6 +225,68 @@ github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801
 │           ├── "Header"
 │           ├── "Query"
 │           └── "Segment"
+├── Authorization: Resource
+│   ├── Owner: AuthorizationProvider
+│   ├── Spec: Object (7 properties)
+│   │   ├── AuthorizationType: *Enum (1 value)
+│   │   │   └── "OAuth2"
+│   │   ├── AzureName: Validated<string> (3 rules)
+│   │   │   ├── Rule 0: MaxLength: 256
+│   │   │   ├── Rule 1: MinLength: 1
+│   │   │   └── Rule 2: Pattern: "^[^*#&+:<>?]+$"
+│   │   ├── Error: *Object (2 properties)
+│   │   │   ├── Code: *string
+│   │   │   └── Message: *string
+│   │   ├── Oauth2GrantType: *Enum (2 values)
+│   │   │   ├── "AuthorizationCode"
+│   │   │   └── "ClientCredentials"
+│   │   ├── Owner: *genruntime.KnownResourceReference
+│   │   ├── Parameters: map[string]string
+│   │   └── Status: *string
+│   └── Status: Object (9 properties)
+│       ├── AuthorizationType: *Enum (1 value)
+│       │   └── "OAuth2"
+│       ├── Conditions: conditions.Condition[]
+│       ├── Error: *Object (2 properties)
+│       │   ├── Code: *string
+│       │   └── Message: *string
+│       ├── Id: *string
+│       ├── Name: *string
+│       ├── Oauth2GrantType: *Enum (2 values)
+│       │   ├── "AuthorizationCode"
+│       │   └── "ClientCredentials"
+│       ├── Parameters: map[string]string
+│       ├── Status: *string
+│       └── Type: *string
+├── AuthorizationProvider: Resource
+│   ├── Owner: Service
+│   ├── Spec: Object (5 properties)
+│   │   ├── AzureName: Validated<string> (3 rules)
+│   │   │   ├── Rule 0: MaxLength: 256
+│   │   │   ├── Rule 1: MinLength: 1
+│   │   │   └── Rule 2: Pattern: "^[^*#&+:<>?]+$"
+│   │   ├── DisplayName: Validated<*string> (2 rules)
+│   │   │   ├── Rule 0: MaxLength: 300
+│   │   │   └── Rule 1: MinLength: 1
+│   │   ├── IdentityProvider: *string
+│   │   ├── Oauth2: *Object (2 properties)
+│   │   │   ├── GrantTypes: *Object (2 properties)
+│   │   │   │   ├── AuthorizationCode: map[string]string
```

Comment:
`AuthorizationCode` and `ClientCredentials` should be secret properties. However, we don't yet support collection of secrets at the moment. @matthchr is working on #3435 which would enable us to mark these properties as secretReferences